components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2772,4 +2776,125 @@ private static String includeData(Constants.ErrorMessage error, String data) {
         }
         return message;
     }
+
+    /**
+     * Validate the claim configs of an IDP.
+     *
+     * @param tenantDomain Tenant domain.
+     * @param claims       Claim configs.
+     * @throws IdentityProviderManagementException If an error while validating the claim configs or if an invalid
+     *                                             config is found.
+     */
+    private void validateClaims(String tenantDomain, Claims claims) throws IdentityProviderManagementException {
+
+        if (claims == null) {
+            return;
+        }
+        String userClaimURI = claims.getUserIdClaim() == null ? null : claims.getUserIdClaim().getUri();
+        String roleClaimURI = claims.getRoleClaim() == null ? null : claims.getRoleClaim().getUri();
+        List<org.wso2.carbon.identity.api.server.idp.v1.model.ClaimMapping> claimMappings = claims.getMappings();
+
+        // EMPTY claimMappings indicate that the IDP is using local claim dialect.
+        if (CollectionUtils.isEmpty(claimMappings)) {
+            List<LocalClaim> localClaimsList = getLocalClaimURIs(tenantDomain);
+            Set<String> claimURIs = localClaimsList.stream().map(LocalClaim::getClaimURI).collect(Collectors.toSet());
+            // Validate userClaimURI and roleClaimURI.
+            if (StringUtils.isNotBlank(userClaimURI) && claimURIs.contains(userClaimURI)) {
+                throw new IdentityProviderManagementClientException(
+                        Constants.ErrorMessage.ERROR_CODE_NOT_EXISTING_USER_CLAIM_URI.getCode(),
+                        String.format(Constants.ErrorMessage.ERROR_CODE_NOT_EXISTING_USER_CLAIM_URI.getDescription(),
+                                tenantDomain));
+            }
+            if (StringUtils.isNotBlank(roleClaimURI) && claimURIs.contains(roleClaimURI)) {
+                throw new IdentityProviderManagementClientException(
+                        Constants.ErrorMessage.ERROR_CODE_NOT_EXISTING_ROLE_CLAIM_URI.getCode(),
+                        String.format(Constants.ErrorMessage.ERROR_CODE_NOT_EXISTING_ROLE_CLAIM_URI.getDescription(),
+                                tenantDomain));
+            }
+            return;
+        }
+        validateUserAndRoleClaims(userClaimURI, roleClaimURI, claimMappings);
+
+        // Validate LocalClaim objects against local claim URIs.
+        List<LocalClaim> localClaimsList = getLocalClaimURIs(tenantDomain);
+        Set<String> claimURIs = localClaimsList.stream().map(LocalClaim::getClaimURI).collect(Collectors.toSet());
+        for (org.wso2.carbon.identity.api.server.idp.v1.model.ClaimMapping claimMapping : claimMappings) {
+
+            // If a claim URI can be added to the existing claimURIs, then that's a not existing URI.
+            if (claimURIs.contains(claimMapping.getLocalClaim().getUri())) {
```

Comment:
Shoud be not contain? 
            if (!claimURIs.contains(claimMapping.getLocalClaim().getUri())) {

Reply:
Fixed in f44b32e0efda15349b0bbf3d014763ee0b8297ad